src/mbgl/style/style_parser.cpp

```diff
@@ -748,12 +748,14 @@ void StyleParser::parseLayout(JSVal value, util::ptr<StyleLayer> &layer) {
     case StyleLayerType::Fill: {
         StyleBucketFill &render = bucket.render.get<StyleBucketFill>();
 
+        parseRenderProperty<VisibilityTypeClass>(value, bucket.visibility, "visibility");
```

Comment:
Instead of repeating this line for each layer type, there can be one line outside the `switch` block, right?